src/sendGCM.js

```diff
@@ -138,7 +138,7 @@ const sendGCM = (regIds, data, settings) => {
     restrictedPackageName: data.restrictedPackageName,
     dryRun: data.dryRun || false,
     data: opts.phonegap === true ? Object.assign(custom, notification) : custom, // See https://github.com/phonegap/phonegap-plugin-push/blob/master/docs/PAYLOAD.md#android-behaviour
-    notification: opts.phonegap === true ? undefined : notification,
+    notification: opts.phonegap === true || data.silent ? undefined : notification,
```

Comment:
please add a test for the correct handling of the `silent` flag in sendGCM

Reply:
please update `README.md` where currently the `silent` flag is declared as a `apn` only parameter.
Ideally add a subsection to the `GCM` section where the "data-only" notifications via `silent=true` are mentioned.

Reply:
@alex-friedl working these updates now. I'm a little confused by the existing silent notification test for GCM. I see that the test for silent push notifications current does this:

```
    it('all responses should be successful (callback, custom data undefined)', (done) => {
      const silentPushData = {
        contentAvailable: true,
        priority: 'normal',
        custom: {
          testKey: 'testValue',
        },
      };
      pn.send(regIds, silentPushData, (err, results) =>
        testSuccess(err, results, done)
      );
    });
```

Is this just assuming that a notification sent without a body and with a custom field is silent without an explicit "silent" flag?

Reply:
@cjroth I actually remember implementing this feature now. Is it not working for you if you set the parameters like in the test here? Shouldn't it be enough to omit the `notification` property when you construct your parameters?

If we still want to move forward with thsi change, we should extend this test and ensure that the `gcm.Message` is constructed as expected if `silent=true` is set.

Thinking about it, it should probably behave consistently with the apns implementation and always set the priority to "normal" if `silent=true`, regardless of what is being passed in the parameters?